src/engine/sim_engine/pool.rs

```diff
@@ -92,21 +92,21 @@ impl Pool for SimPool {
     }
 
     fn create_filesystems<'a, 'b>(&'a mut self,
-                                  specs: &[&'b str])
+                                  specs: &[(&'b str, Option<Sectors>)])
                                   -> EngineResult<Vec<(&'b str, FilesystemUuid)>> {
-        let names: HashSet<_, RandomState> = HashSet::from_iter(specs);
-        for name in &names {
+        let names: HashMap<_, _> = HashMap::from_iter(specs.iter().map(|&tup| (tup.0, tup.1)));
```

Comment:
fwiw I played around with that but couldn't get it to work, b/c it's a &tuple instead of just a tuple.